demos/common/cpp/models/src/detection_model_retinaface_pt.cpp

```diff
@@ -0,0 +1,250 @@
+    // --------------------------- Configure input & output -------------------------------------------------
+    // --------------------------- Prepare input blobs ------------------------------------------------------
+    slog::info << "Checking that the inputs are as the demo expects" << slog::endl;
+    InferenceEngine::InputsDataMap inputInfo(cnnNetwork.getInputsInfo());
+    if (inputInfo.size() != 1) {
+        throw std::logic_error("This demo accepts networks that have only one input");
+    }
+    InferenceEngine::InputInfo::Ptr& input = inputInfo.begin()->second;
+    std::string imageInputName = inputInfo.begin()->first;
+    inputsNames.push_back(imageInputName);
+    input->setPrecision(InferenceEngine::Precision::U8);
+    if (useAutoResize) {
+        input->getPreProcess().setResizeAlgorithm(InferenceEngine::ResizeAlgorithm::RESIZE_BILINEAR);
+        input->getInputData()->setLayout(InferenceEngine::Layout::NHWC);
+    }
+    else {
+        input->getInputData()->setLayout(InferenceEngine::Layout::NCHW);
+    }
+
+    //--- Reading image input parameters
+    imageInputName = inputInfo.begin()->first;
+    const InferenceEngine::TensorDesc& inputDesc = inputInfo.begin()->second->getTensorDesc();
+    netInputHeight = getTensorHeight(inputDesc);
+    netInputWidth = getTensorWidth(inputDesc);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    slog::info << "Checking that the outputs are as the demo expects" << slog::endl;
+
+    InferenceEngine::OutputsDataMap outputInfo(cnnNetwork.getOutputsInfo());
+    landmarksNum = 0;
+
+    std::vector<uint32_t> outputsSizes[OT_MAX];
+    for (auto& output : outputInfo) {
+        output.second->setPrecision(InferenceEngine::Precision::FP32);
+        output.second->setLayout(output.second->getDims().size()==4 ? InferenceEngine::Layout::NCHW : InferenceEngine::Layout::CHW);
+        outputsNames.push_back(output.first);
+
+        outputsNames.resize(2);
+        if (output.first.find("bbox") != std::string::npos) {
+            outputsNames[OT_BBOX] = output.first;
+        }
+        else if (output.first.find("cls") != std::string::npos) {
+            outputsNames[OT_SCORES] = output.first;
+        }
+        else if (output.first.find("landmark") != std::string::npos) {
+            outputsNames.resize(std::max(outputsNames.size(), (size_t)OT_LANDMARK + 1));
+            outputsNames[OT_LANDMARK] = output.first;
+            landmarksNum = output.second->getDims()[2]/2; // Each landmark consist of 2 variables (x and y)
+        }
+        else {
+            continue;
+        }
+    }
+
+    if (outputsNames[OT_BBOX] == "" || outputsNames[OT_SCORES] == "") {
+        throw std::runtime_error("Bbox or cls layers are not found");
+    }
+
+    if (outputsNames.size() != 2 && outputsNames.size() != 3) {
+        throw std::logic_error("Expected 2 or 3 output blobs");
+    }
```

Comment:
The `outputsNames.size` will be always 2 or 3.

Reply:
removed. Assuming that we just skip extra outputs if they are present. But I kept check for outputs names